src/main/scala/is/hail/variant/VariantDataset.scala

```diff
@@ -1186,6 +1187,58 @@ class VariantDatasetFunctions(private val vds: VariantSampleMatrix[Genotype]) ex
     ret
   }
 
+
+  def queryGenotypes(expr: String): (Annotation, Type) = {
+    val qv = queryGenotypes(Array(expr))
+    assert(qv.length == 1)
+    qv.head
+  }
+
+  def queryGenotypes(exprs: Array[String]): Array[(Annotation, Type)] = {
+    val aggregationST = Map(
+      "global" -> (0, vds.globalSignature),
+      "g" -> (1, TGenotype),
+      "v" -> (2, TVariant),
+      "va" -> (3, vds.vaSignature),
+      "s" -> (4, TSample),
+      "sa" -> (5, vds.saSignature))
+    val ec = EvalContext(Map(
+      "global" -> (0, vds.globalSignature),
+      "gs" -> (1, TAggregable(TGenotype, aggregationST))))
+
+    val ts = exprs.map(e => Parser.parseExpr(e, ec))
+
+    val localGlobalAnnotation = vds.globalAnnotation
+    val (zVal, seqOp, combOp, resOp) = Aggregators.makeFunctions[Genotype](ec, { case (ec, g) =>
+      ec.set(1, g)
+    })
+
+    val sampleIdsBc = vds.sampleIdsBc
+    val sampleAnnotationsBc = vds.sampleAnnotationsBc
+    val globalBc = vds.sparkContext.broadcast(vds.globalAnnotation)
+
+    val result = vds.rdd.mapPartitions { it =>
+      val zv = zVal.map(_.copy())
+      ec.set(0, globalBc.value)
+      it.foreach { case (v, (va, gs)) =>
+        var i = 0
+        ec.set(2, v)
+        ec.set(3, va)
+        gs.foreach { g =>
+          ec.set(4, sampleIdsBc.value(i))
+          ec.set(5, sampleAnnotationsBc.value(i))
+          seqOp(zv, g)
+          i += 1
+        }
+      }
+      Iterator(zv)
+    }.fold(zVal.map(_.copy()))(combOp)
```

Comment:
This is the result after fixing that!  I do want to note, though, that the following code does _NOT_ work:

```scala
vds.rdd.treeAggregate(zVal)(seqOp, combOp)
```

...because the treeAggregate does a serialize/deserialize on the zero value, which contain array references.  This is a bit confusing, but I confirmed it a few times.